postprocessing/pp-avformat.c

```diff
@@ -0,0 +1,92 @@
+/*! \file    pp-avformat.c
+ * \copyright GNU General Public License v3
+ *
+ * \ingroup postprocessing
+ * \ref postprocessing
+ */
+
+#include "pp-avformat.h"
+
+void janus_pp_setup_avformat(void) {
+	/* Setup FFmpeg */
+#if ( LIBAVFORMAT_VERSION_INT < AV_VERSION_INT(58,9,100) )
+	av_register_all();
+#endif
+	/* Adjust logging to match the postprocessor's */
+	av_log_set_level(janus_log_level <= LOG_NONE ? AV_LOG_QUIET :
+		(janus_log_level == LOG_FATAL ? AV_LOG_FATAL :
+			(janus_log_level == LOG_ERR ? AV_LOG_ERROR :
+				(janus_log_level == LOG_WARN ? AV_LOG_WARNING :
+					(janus_log_level == LOG_INFO ? AV_LOG_INFO :
+						(janus_log_level == LOG_VERB ? AV_LOG_VERBOSE : AV_LOG_DEBUG))))));
+
+}
+
+AVFormatContext *janus_pp_create_avformatcontext(const char *format, const char *metadata, const char *destination) {
+	janus_pp_setup_avformat();
+
+	AVFormatContext *ctx = avformat_alloc_context();
+	if(!ctx)
+		return NULL;
+
+	/* We save the metadata part as a comment (see #1189) */
+        if(metadata)
+                av_dict_set(&ctx->metadata, "comment", metadata, 0);
```

Comment:
Broken indentation in the above lines: per the project code style, we use tabs, not spaces.